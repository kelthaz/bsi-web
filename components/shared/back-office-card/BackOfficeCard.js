import PropTypes from 'prop-types';
import SvgAgendarCita from '../../svgs/SvgAgendarCita';
import SvgListCheckBackOffice from '../../svgs/SvgListCheckBackOffice';
import styles from './back-office-card.module.scss';

const BackOfficeCard = ({ iconoAgendar, title, subTitle, haveButton }) => (
  <div className={`row ${iconoAgendar === true ? 'card-simple-blue-sky' : 'card-simple-blue-morning'}`}>
    <div className="col-2 mt-4">{iconoAgendar === true ? <SvgAgendarCita /> : <SvgListCheckBackOffice />}</div>
    <div className="col-7 mt-4 pr-0 pl-1">
      <h4 className="body1">{title}</h4>
      <div className="body3">{subTitle}</div>
    </div>
    {haveButton === true ? (
      <div className={`col-3 mt-5 px-0 ${styles.button}`}>
        <button className="btn-link-arrow-right-inverted" type="button">
          Ver detalle
        </button>
      </div>
    ) : null}
  </div>
);

BackOfficeCard.propTypes = {
  iconoAgendar: PropTypes.bool,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  haveButton: PropTypes.bool,
};

BackOfficeCard.defaultProps = {
  iconoAgendar: true,
  title: null,
  subTitle: null,
  haveButton: true,
};

export default BackOfficeCard;
