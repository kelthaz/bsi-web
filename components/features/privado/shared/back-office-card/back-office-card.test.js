import { shallow } from 'enzyme';
import React from 'react';
import BackOfficeCard from './BackOfficeCard';

describe('Pruebas en el componente cards del backOffice', () => {
  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const props = { iconoAgendar: true, title: 'Prueba del titulo', subTitle: 'Prueba del subtitle' };
    // act
    const wrapper = shallow(<BackOfficeCard {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });
});
