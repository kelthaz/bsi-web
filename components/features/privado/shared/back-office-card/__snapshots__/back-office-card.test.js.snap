// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente cards del backOffice Debe de mostrarse correctamente el componente 1`] = `
<div
  className="row card-simple-blue-sky"
>
  <div
    className="col-2 mt-4"
  >
    <SvgAgendarCita />
  </div>
  <div
    className="col-7 mt-4 pr-0 pl-1"
  >
    <h4
      className="body1"
    >
      Prueba del titulo
    </h4>
    <div
      className="body3"
    >
      Prueba del subtitle
    </div>
  </div>
  <div
    className="col-3 mt-5 px-0 button"
  >
    <button
      className="btn-link-arrow-right-inverted"
      type="button"
    >
      Ver detalle
    </button>
  </div>
</div>
`;
